backend/users/views.py

```diff
@@ -1,4 +1,37 @@
-from django.shortcuts import render  # noqa
+from django.contrib.auth import authenticate, login
+from django.contrib.auth.views import LoginView, LogoutView
+from django.http import HttpResponseRedirect
+from django.urls import reverse_lazy
+from django.views.generic import CreateView
 
+from users.forms import UserSignUpForm
+from users.models import User
 
-# Create your views here.
+
+class UserLoginView(LoginView):
+    template_name = "users/login.html"
+
+
+class UserLogoutView(LogoutView):
+    template_name = "users/logout.html"
+
+
+class UserSignUpView(CreateView):
+    model = User
+    form_class = UserSignUpForm
+    template_name = "users/signup.html"
+    success_url = reverse_lazy("home")
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.request.user.is_authenticated:
+            context["user_already_logged"] = True
```

Comment:
You can remove this line here (and use `user.is_authenticated` directly in the template) for the same effect.

The whole `get_context_data` method can go, as there won't be any custom behaviors implemented after that.

Reply:
I may have missed other uses of `user_already_logged` across the project, so it would be interesting to search for it and also change.